gcp/api/server.py

```diff
@@ -992,7 +998,9 @@ def query_by_generic_helper(results: list, cursor, context: QueryContext,
   query: ndb.Query = base_query.filter(osv.Bug.affected_fuzzy == version)
   it: ndb.QueryIterator = query.iter(start_cursor=context.page_token)
   while (yield it.has_next_async()):
-    if len(results) >= context.total_responses.page_limit():
+    now = datetime.now()
```

Comment:
nit: is it possible to write some kind of wrapper such that there's less  duplication of  this logic across all the places we iterate?